src/Compilers/VisualBasic/Portable/Symbols/ConstraintsHelper.vb

```diff
@@ -1086,28 +1091,55 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             Return New CompoundDiagnosticInfo(diagnostics)
         End Function
 
+        Public Enum ConstructorConstraintError
+            None
+            NoPublicParameterlessConstructor
+            MustInheritType
+            HasRequiredMembers
+        End Enum
+
         ''' <summary>
         ''' Return true if the class type has a public parameterless constructor.
         ''' </summary>
-        Public Function HasPublicParameterlessConstructor(type As NamedTypeSymbol) As Boolean
+        Public Function HasPublicParameterlessConstructor(type As NamedTypeSymbol, synthesizedIfMissing As Boolean) As ConstructorConstraintError
             type = type.OriginalDefinition
-            Debug.Assert(type.TypeKind = TypeKind.Class)
+            Debug.Assert(type.TypeKind = TypeKind.Class OrElse type.TypeKind = TypeKind.Structure)
 
             Dim sourceNamedType = TryCast(type, SourceNamedTypeSymbol)
```

Comment:
>Dim sourceNamedType = TryCast(type, SourceNamedTypeSymbol) [](http://example.com/codeflow?start=12&length=58)

This is a preexisting condition, but it looks like we are not dealing with constructed types properly here. Consider opening an issue to follow up.

Reply:
[https://github.com/dotnet/roslyn/issues/66564](https://github.com/dotnet/roslyn/issues/66564)